### ISSUE TITLE
T::Struct doesn't work on Ruby 2.3 and 2.4 any more after #2336

### ISSUE THREAD

**Issue by manhhung741** [sorbet/sorbet#2487]
#### Input

```ruby
class Example < T::Struct
  const :name, String
end
```

#### Observed output

```
Failure/Error: const :name, String
NoMethodError:
  private method `attr_reader' called for WizardName:Class
  Did you mean?  cattr_reader
                 mattr_reader
```

#### Expected behavior

It shouldn't error just defining a T::Struct. I found the blame is #2336

https://github.com/sorbet/sorbet/blame/master/gems/sorbet-runtime/lib/types/props/decorator.rb#L532

**Comment by jez** [comment]
Thanks!

Open to a PR that uses `.send(:attr_reader)` here:

https://github.com/sorbet/sorbet/pull/2336/files#diff-296495687573a8a72557282b673db5f5R514

**Comment by manhhung741** [comment]
Thanks you @jez!

**Comment by djudd-stripe** [comment]
Closing this since https://github.com/sorbet/sorbet/pull/2509 fixed it (though IIUC 2.3 is still broken due to https://github.com/sorbet/sorbet/issues/2556)
